plugins/modules/rpm_distribution.py

```diff
@@ -112,6 +127,7 @@ def main():
             "repository": {},
             "content_guard": {},
             "generate_repo_config": {"type": "bool"},
+            "pulp_labels": {"type": "dict"},
```

Comment:
There's a way to specify this as a dictionary of strings. Let's do that.

Reply:
Hey @mdellweg. I've looked into this and cannot determine the method you describe. Specifically, it doesn't appear that `argument_spec` allows for arbitrary keys, which `pulp_labels` does. I could potentially change `pulp_labels` to a list of dictionaries, but that seems generally unhelpful to me. Could you provide some additional guidance if you know of a better way?

[Here is some discussion I found](http://webcache.googleusercontent.com/search?q=cache%3Ahttps%3A%2F%2Fgroups.google.com%2Fg%2Fansible-devel%2Fc%2FyYRWbLH5Zc4&sca_esv=b8c8c187ff72a731&source=hp&ei=Wb-wZrnnCOer5NoPu6nUyAI&iflsig=AL9hbdgAAAAAZrDNaShwIneiR1si3UODdgHyjj-su04t&ved=0ahUKEwj59_ed592HAxXnFVkFHbsUFSkQ4dUDCBg&uact=5&oq=cache%3Ahttps%3A%2F%2Fgroups.google.com%2Fg%2Fansible-devel%2Fc%2FyYRWbLH5Zc4&gs_lp=Egdnd3Mtd2l6Ij1jYWNoZTpodHRwczovL2dyb3Vwcy5nb29nbGUuY29tL2cvYW5zaWJsZS1kZXZlbC9jL3lZUldiTEg1WmM0Mg0QABiABBixAxhGGPkBMggQLhiABBixAzIIEAAYgAQYsQMyERAuGIAEGLEDGIMBGMcBGK8BMggQABiABBixAzIXEC4YgAQYsQMYgwEYxwEYmAUYmgUYrwEyBRAAGIAEMgUQABiABDIIEC4YgAQYsQMyCBAAGIAEGLEDSNwPUABYzQtwAHgAkAEAmAGBAaAB9QSqAQMyLjS4AQPIAQD4AQH4AQKYAgagAp8FwgIREC4YgAQYsQMY0QMYgwEYxwHCAgsQLhiABBixAxiDAcICDhAuGIAEGLEDGNEDGMcBwgILEC4YgAQY0QMYxwHCAg4QABiABBixAxiDARiKBcICCxAuGIAEGMcBGK8BwgIOEC4YgAQYsQMYgwEYigXCAgoQABiABBixAxgKwgIFEC4YgATCAgsQLhiABBixAxjUApgDAJIHAzEuNaAH1VU&sclient=gws-wiz), for reference.

Reply:
Have you tried "elements" or does that only work with "list"?

Reply:
Unfortunately, that is only for use with the `list` type. I did give it a try just for fun, but it seems documentation does in fact match implementation on this one.

```
"msg": "Invalid type dict for option '{'key1': 'value1', 'key2': 'value2'}', elements value check is supported only with 'list' type"
```